public/app/plugins/datasource/elasticsearch/hooks/useFields.ts

```diff
@@ -8,27 +11,38 @@ import {
 
 type AggregationType = BucketAggregationType | MetricAggregationType;
 
-const getFilter = (aggregationType: AggregationType) => {
-  // For all metric types we want only numbers, except for cardinality
+const getFilter = (type: AggregationType | string[]) => {
   // TODO: To have a more configuration-driven editor, it would be nice to move this logic in
   // metricAggregationConfig and bucketAggregationConfig so that each aggregation type can specify on
   // which kind of data it operates.
-  if (isMetricAggregationType(aggregationType)) {
-    if (aggregationType !== 'cardinality') {
-      return 'number';
-    }
+  if (Array.isArray(type)) {
+    return type;
+  }
 
-    return void 0;
+  if (isMetricAggregationType(type)) {
+    switch (type) {
+      case 'cardinality':
+        return void 0;
+      case 'top_metrics':
+        // FIXME: DOUBLE CHECK THIS
+        return ['string', 'boolean', 'number', 'ip'];
```

Comment:
```suggestion
        // top_metrics was introduced in 7.7 where `metrics` only supported number:
        // https://www.elastic.co/guide/en/elasticsearch/reference/7.11/search-aggregations-metrics-top-metrics.html#_metrics
        // TODO: starting from 7.11 it supports ips and keywords as well:
        // https://www.elastic.co/guide/en/elasticsearch/reference/7.7/search-aggregations-metrics-top-metrics.html#_metrics
        return ['number'];
```

does it make sense?

we can add support for other fields in a follow-up PR, there's some refactoring needed here in order to select fields based on configured ES version